### PR TITLE
Docs: Caching

### PR DESCRIPTION
**Pull request by Rachael-Graham**
# Description

Docs for deploying caching server in 1.12 and configuring a listener to cache.

### CODE REVIEW

File: docs/content/guides/traffic_management/listener_configuration/caching.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+Cache responses from upstream services by deploying a caching server for your Gloo Edge setup and applying a caching filter to a listener.
+
+{{% notice note %}}
+This feature is available only for Gloo Edge Enterprise v1.12.x and later.
+{{% /notice %}}
+
+The Gloo Edge Enterprise caching filter is an extension (implementing filter) of the [Envoy cache filter](https://www.envoyproxy.io/docs/envoy/latest/start/sandboxes/cache) and takes advantage of all the cache-ability checks that are applied. However, Gloo Edge also provides the ability to store the cached objects in a Redis instance, including Redis configuration options such as setting a password.
+
+When you enable caching during installation, the caching server deployment is automatically created for you and is managed by Gloo Edge. Then, you can configure an HTTP or HTTPS listener to cache responses for its upstream services. When the listener routes a request comes to an upstream, the response from the upstream is automatically cached for one hour. All subsequent requests receive the cached response.
```
Comment (Nadine2016) [inline docs/content/guides/traffic_management/listener_configuration/caching.md:15]:
```suggestion
When you enable caching during installation, the caching server deployment is automatically created for you and is managed by Gloo Edge. Then, you can configure an HTTP or HTTPS listener to cache responses for its upstream services. When the listener routes a request to an upstream, the response from the upstream is automatically cached for one hour. All subsequent requests receive the cached response.
```

File: docs/content/guides/traffic_management/listener_configuration/caching.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+   * Caching server: Set `global.extensions.caching.enabled: true` to enable the caching server deployment.
+   * Redis overrides: By default, the caching server uses the Redis instance that is deployed with Gloo Edge. To use your own Redis instance, such as in production deployments:
+     * Set `redis.disabled` to `true` to disable the default Redis instance.
+     * Set `redis.service.name` to the name of the Redis service instance. If the instace is an external service, set the endpoint of the external service as the value.
+     * For other Redis override settings, see the Redis section of the [Enterprise Helm chart values]({{% versioned_link_path fromRoot="/reference/helm_chart_values/enterprise_helm_chart_values/" %}}).
+
+2. Verify that the caching server is deployed.
+   ```sh
+   kubectl --namespace gloo-system get all | grep caching
+   ```
```
Comment (Nadine2016) [inline docs/content/guides/traffic_management/listener_configuration/caching.md:31]: maybe add an example output here

File: docs/content/guides/traffic_management/listener_configuration/caching.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+     bindAddress: ‘::’
+     bindPort: 8080
+     proxyNames:
+     - gateway-proxy
+     httpGateway:
+       options:
+         caching:
+           cachingServiceRef:
+             name: caching-service
+             namespace: gloo-system
```
Comment (Rachael-Graham) [inline docs/content/guides/traffic_management/listener_configuration/caching.md:72]: Is this the correct listener configuration for caching?

File: docs/content/guides/traffic_management/listener_configuration/caching.md [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+           cachingServiceRef:
+             name: caching-server
+             namespace: gloo-system
+     options:
+       accessLoggingService:
+         accessLog:
+         - fileSink:
+             path: /dev/stdout
+             stringFormat: (specific log format that would indicate responses are being cached...?)
+   {{< /highlight >}}
```
Comment (Rachael-Graham) [inline docs/content/guides/traffic_management/listener_configuration/caching.md:96]: To verify that responses are being cached, I was wondering if users might be able to check out the gateway-proxy logs to see if the responses are different? 

e.g., on the first request, the log would indicate something different than the logs for subsequent requests bc subsequent requests are getting cached responses

**Review by artberger — DISMISSED** [review]

File: docs/content/guides/traffic_management/listener_configuration/caching.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+Use the Gloo Edge Enterprise caching server to cache responses from upstream services for quicker response times.
+
+{{% notice note %}}
+This feature is available only for Gloo Edge Enterprise v1.12.x and later.
+{{% /notice %}}
+
+The Gloo Edge Enterprise caching filter is an extension (implementing filter) of the [Envoy cache filter](https://www.envoyproxy.io/docs/envoy/latest/start/sandboxes/cache) and takes advantage of all the cache-ability checks that are applied. However, Gloo Edge also provides the ability to store the cached objects in a Redis instance, including Redis configuration options such as setting a password.
+
+When you enable caching during installation, the caching server deployment is automatically created for you and is managed by Gloo Edge. Then, you can configure an HTTP or HTTPS listener to cache responses for its upstream services. When the listener routes a request to an upstream, the response from the upstream is automatically cached for one hour. All subsequent requests receive the cached response.
```
Comment (Rachael-Graham) [inline docs/content/guides/traffic_management/listener_configuration/caching.md:15]:
```suggestion
When you enable caching during installation, the caching server deployment is automatically created for you and is managed by Gloo Edge. Then, you must configure an HTTP or HTTPS listener to cache responses for its upstream services. When the listener routes a request to an upstream, the response from the upstream is automatically cached for one hour. All subsequent requests receive the cached response.
```

File: docs/content/guides/traffic_management/listener_configuration/caching.md [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+<!-- future work
+## Configure settings for the caching server
+
+should be able to configure general settings for the server in the future, like the default caching time
+
+https://docs.solo.io/gloo-edge/master/reference/api/github.com/solo-io/gloo/projects/gloo/api/v1/enterprise/options/caching/caching.proto.sk/#settings
+-->
+
+## Configure caching for a listener
+
```
Comment (Rachael-Graham) [inline docs/content/guides/traffic_management/listener_configuration/caching.md:59]:
```suggestion
## Configure caching for a listener

After you enable the caching server, you must configure caching on a listener defined in a Gateway CRD.
```

**Review by artberger — APPROVED** [review]

**Pull request merged**